docs/_docs/07-dg-code-structure.md

```diff
@@ -6,3 +6,28 @@ last_modified_at: 2020-06-20
 toc: true
 ---
 
+Hyperspace code is written in Scala and it is using 'sbt' as the build tool. Below is the list of folders in the code base hierarchy:
+### docs
+“docs” folder should be the starting point for looking into Hyperspace code base. It contains useful documentations on different aspects of Hyperspace project, including coding and formatting guidelines, Hyperspace configurations and information on how to build the project in various environments. 
+### src
+“src” folder contains Hyperspace’s source and test code. 
+The source code is available under:
+* src/main/scala/com/microsoft/hyperspace
+
+The source code includes implementation of core functionality and features of Hyperspace and is organized in different packages:
+- **actions** : It includes the implementation of Hyperspace actions to create, maintain and remove indexes in a concurrent manner.
+- **index**: It includes implementation of Hyeprspace indexes, optimizations rules and other core features. 
+- **util**: Various utility codes that are used by different features and components are added here.
+
+If you are working on adding changes to Hyeprspace, you would need to add or modify code in above packages.
+
+The test code is available under:
+* src/test/scala/com/microsoft/hyperspace
+
+The test code has a similar structure as the source code and each package in the test code includes unit test cases for the corresponding feature or functionality from the source code. Any code change or additional feature implementation needs to add relevant cases which fully cover the modified code.
+### dev
+“dev” folder contains required resources for contributing code to the Hyperspace project as a developer. Currently, it includes the Scala formatting configuration file that is needed to make sure any new code change complies with the Hyperspace’s coding guidelines.
+### build
+Hyperspace uses “sbt” is the build tool for building its code. “build” folder contains sbt build scripts and configurations which are used to create an artifact from Hyperspace code.
+### project
+“project” folder contains additional sbt build properties and plugin configurations which are needed to correctly format Hyperspace code and build it to generate a Hyperspace artifact.
```

Comment:
`“project” folder contains additional sbt build properties and plugin configurations.` is good enough.

Reply:
Done.